pkg/scheduler/framework/runtime/framework.go

```diff
@@ -417,16 +417,43 @@ func getScoreWeights(f *frameworkImpl, pluginsMap map[string]framework.Plugin, p
 	return nil
 }
 
+type orderedSet struct {
+	set  sets.String
+	list []string
+}
+
+func newOrderedSet() *orderedSet {
+	return &orderedSet{set: sets.NewString()}
+}
+
+func (os *orderedSet) insert(s string) {
+	os.set.Insert(s)
+	os.list = append(os.list, s)
+}
+
+func (os *orderedSet) has(s string) bool {
+	return os.set.Has(s)
+}
+
+func (os *orderedSet) delete(i int, s string) {
```

Comment:
Small suggestion, it seems a bit strange that the user needs to know the index of the item to be deleted. This feels a bit inconsistent with the semantics of set.  

How about use a map `map[string]int` to replace `set` here?

Reply:
sgtm. Thanks.